test/python/transpiler/test_preset_passmanagers.py

```diff
@@ -919,3 +919,37 @@ def test_input_dag_copy(self):
         qc.cx(1, 0)
         circ = transpile(qc, basis_gates=["u3", "cz"])
         self.assertIsInstance(circ, QuantumCircuit)
+
+
+@ddt
+class TestOptimizationOnSize(QiskitTestCase):
+    """Test the optimization levels for optimization based on
+    both size and depth of the circuit.
+    See https://github.com/Qiskit/qiskit-terra/pull/7542
+    """
+
+    @data(2, 3)
+    def test_size_optimization(self, level):
+        """Test the levels for optimization based on size of circuit"""
+        qc = QuantumCircuit(8)
+        qc.cx(1, 2)
+        qc.cx(2, 3)
+        qc.cx(5, 4)
+        qc.cx(6, 5)
+        qc.cx(4, 5)
+        qc.cx(3, 4)
+        qc.cx(5, 6)
+        qc.cx(5, 4)
+        qc.cx(3, 4)
+        qc.cx(2, 3)
+        qc.cx(1, 2)
+        qc.cx(6, 7)
+        qc.cx(6, 5)
+        qc.cx(5, 4)
+        qc.cx(7, 6)
+        qc.cx(6, 7)
+
+        circ = transpile(qc, optimization_level=level)
+
+        self.assertLessEqual(circ.size(), qc.size())
+        self.assertLessEqual(circ.depth(), qc.depth())
```

Comment:
```suggestion
        self.assertLess(circ.size(), qc.size())
        self.assertLessEqual(circ.depth(), qc.depth())
```
We should make sure that the test captures the case we're trying to fix here, so the output size should be strictly less than the input. We could also check that there are no gates in the output circuit on `q0` through `q3`.